# Practice - Explorin Datasets without Pandas

The following code is mostly based on exercises from Dataquest. The main purpose of this notebook is to get comfortable with exploring several datasets without using pandas.

#### Business Problem: The aim of this analysis is to help developers understand what type of apps are likely to attract more users on Google Play and the App Store

The following datasets are available online:
  - Apple: https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps
  - Google: https://www.kaggle.com/lava18/google-play-store-apps

## Data Exploration

In [14]:
# open the 2 files
opened_file_apple = open('AppleStore.csv', encoding='utf8')
opened_file_google = open('googleplaystore.csv', encoding = 'utf8')

from csv import reader
read_file = reader(opened_file_apple)
read_file_g = reader(opened_file_google)

apple_data = list(read_file)
google_data = list(read_file_g)


In [33]:
# open and explore documents
def explore_data(dataset, start, end, rows_and_columns = False):
    dataset_slice = dataset[start:end]
    for row in dataset_slice:
        print(row)
        print('\n')
        
    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

### Apple Data

In [25]:
explore_data(apple_data,2,3,4)

['2', '281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1']


Number of rows: 7198
Number of columns: 17


In [20]:
#header
apple_data[0]

['',
 'id',
 'track_name',
 'size_bytes',
 'currency',
 'price',
 'rating_count_tot',
 'rating_count_ver',
 'user_rating',
 'user_rating_ver',
 'ver',
 'cont_rating',
 'prime_genre',
 'sup_devices.num',
 'ipadSc_urls.num',
 'lang.num',
 'vpp_lic']

### Google data

In [36]:
explore_data(google_data,2,3,4)

['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


Number of rows: 10842
Number of columns: 13


In [22]:
# header google
google_data[0]

['App',
 'Category',
 'Rating',
 'Reviews',
 'Size',
 'Installs',
 'Type',
 'Price',
 'Content Rating',
 'Genres',
 'Last Updated',
 'Current Ver',
 'Android Ver']

### Analysis

Accoridng to the two datasets, we seem to have more datapoints for Google than Apple (7198 and 10842, respectively). When comparing the two datasets, some features will definetely contribute more to the purpose of this study: which apps customers prefer.
On Apple: Track Name, Price, Rating_count_lot, Prime_genre
On Google: App, Rating, Price, Category, Install, Genres

## Data Cleaning

### Apple

In [78]:
for row in apple_data:
    header_length = len(apple_data[0])
    rowlength = len(row) 
    if rowlength != header_length:
        print(row)
        print(apple_data.index(row))

In [79]:
print(row)
print(apple_data.index(row))

['11097', '1188375727', 'Escape the Sweet Shop Series', '90898432', 'USD', '0', '3', '3', '5', '5', '1.0', '4+', 'Games', '40', '0', '2', '1']
7197


### Google

In [80]:
for row in google_data:
    header_length = len(google_data[0])
    rowlength = len(row) 
    if rowlength != header_length:
        print(row)
        print(google_data.index(row))

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']
10473
